### PR TITLE
Attempt at warnings.

### PR DESCRIPTION
**Pull request by amitjamadagni**
Attempt at fixing some of the warnings.

### CODE REVIEW

**Comment by acroy** [comment]
This looks good to me. Could you try to address also the "Union" warning (probably using `Compat`)?

**Comment by amitjamadagni** [comment]
@acroy there seems to be a  problem with methods of the form`function(arg::@compat Union{arg_1...}...)` leading to seg fault on the nightly release.

File: src/arrays/quarray.jl [hunk 1]
```diff
@@ -153,12 +153,12 @@
     typealias TupleArray{T<:Tuple,N} Array{T,N}
 
     # redudant definitions added to resolve ambiguity warnings
-    Base.getindex{B<:LabelBasis}(larr::DualVector{B}, tups::Union(Tuple,TupleArray)) = getindex(larr, bases(larr,1)[tups])
-    Base.getindex{B<:LabelBasis}(larr::CTranspose{B}, tups::Union(Tuple,TupleArray)...) = getindex(larr, map(getindex, bases(larr), tups)...)
+    Base.getindex{B<:LabelBasis}(larr::DualVector{B}, tups::@compat Union{Tuple,TupleArray}) = getindex(larr, bases(larr,1)[tups])
+    Base.getindex{B<:LabelBasis}(larr::CTranspose{B}, tups::@compat Union{Tuple,TupleArray}...) = getindex(larr, map(getindex, bases(larr), tups)...)
```
Comment (acroy) [inline src/arrays/quarray.jl]: You think it is this line? Try `@compat(Union{Tuple,TupleArray})...` instead ...

**Comment by acroy** [comment]
To get rid of the last warning, it might be sufficient to move the conflicting line 177 before line 175 in quarray.jl?

**Comment by acroy** [comment]
Looks good to me. @jrevels : Any idea regarding the last warning?

**Comment by jrevels** [comment]
You mean:

```julia
julia> using QuBase
WARNING: New definition
    convert(Type{#QV<:QuBase.AbstractQuArray{B<:QuBase.AbstractBasis, T<:Any, 1}}, #DV<:QuBase.CTranspose{B<:QuBase.AbstractBasis, T<:Any, 1, Q<:Any}) at /Users/jarrettrevels/.julia/QuBase/src/arrays/quarray.jl:175
is ambiguous with:
    convert(Type{#T<:Any}, #T<:Any) at essentials.jl:59.
To fix, define
    convert(Type{_<:QuBase.CTranspose{B<:QuBase.AbstractBasis, T<:Any, 1, Q<:Any}}, _<:QuBase.CTranspose{B<:QuBase.AbstractBasis, T<:Any, 1, Q<:Any})
before the new definition.
```

I'd just remove line 175. It's not really doing much anyway. The conversion zoo can likely be made simpler if we make the move to using traits instead of type aliases for identifying states/operators/etc.

**Comment by acroy** [comment]
@jrevels : Ok, you are probably right. It is not worth to fight for this error message.
@amitjamadagni : Could you remove the offending line?

**Comment by amitjamadagni** [comment]
@acroy this is also ready !

**Comment by acroy** [comment]
Nice!

**Pull request merged**